diff a/configure b/configure
--- a/configure
+++ b/configure
@@ -294,7 +294,7 @@
     # If findlib has been configured -sitelib $(ocamlc -where) then there's
     # nothing to do, but otherwise we need to put OCaml's Standard Library
     # into the path setting.
-    if [ ! -e "${ocaml_sitelib}/stdlib/META" ]; then
+    if [ ! -e "${ocaml_sitelib}/stdlib.cmi" ]; then
         ocamlpath="${ocaml_core_stdlib}${path_sep}${ocamlpath}"
     fi
 fi
