--- ./src/migrate_parsetree_driver.ml
+++ ./src/migrate_parsetree_driver.ml
@@ -371,9 +371,14 @@ let load_file file =
       | Intf fn ->
         (fn, Intf (Parse.interface lexbuf)))
 
-let with_output output ~f =
+let with_output ?bin output ~f =
   match output with
-  | None -> f stdout
+  | None ->
+      begin match bin with
+      | Some bin -> set_binary_mode_out stdout bin
+      | None -> ()
+      end;
+      f stdout
   | Some fn -> with_file_out fn ~f
 
 type output_mode =
@@ -416,7 +421,7 @@ let process_file ~config ~output ~output_mode ~embed_errors file =
   in
   match output_mode with
   | Dump_ast ->
-    with_output output ~f:(fun oc ->
+    with_output ~bin:true output ~f:(fun oc ->
       let ast =
         match ast with
         | Intf (_, sg) -> Ast_io.Intf ((module OCaml_current), sg)
