From 0cf3c6ad7ce2a2b2806faceccfb0a9321da5e22a Mon Sep 17 00:00:00 2001
From: David Allsopp <david.allsopp@metastack.com>
Date: Fri, 26 Jul 2019 12:12:19 +0100
Subject: [PATCH] Don't build manpages for stdlib docs
---
 ocamldoc/Makefile            | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ocamldoc/Makefile b/ocamldoc/Makefile
index b109815071..e31e441f61 100644
--- a/ocamldoc/Makefile
+++ b/ocamldoc/Makefile
@@ -170,7 +170,7 @@ LIBCMIFILES = $(LIBCMOFILES:.cmo=.cmi)
 
 
 .PHONY: all
-all: lib exe generators manpages
+all: lib exe generators
 
 manpages: generators
 
-- 
2.20.1

