--- ./Makefile
+++ ./Makefile
@@ -59,13 +59,8 @@
 
 # If the compiler is MSVC, then object file names end in .obj instead of .o.
 
-ifneq (,$(shell ocamlc -config | grep -E "ccomp_type: msvc"))
-  OBJ          := obj
-# LIBSUFFIX    := lib
-else
-  OBJ          := o
-# LIBSUFFIX    := a
-endif
+OBJ          := $(shell ocamlc -config | awk -F '[\t \r]+' '/^ext_obj:/ {print $$2}' | tr -d '.')
+#LIBSUFFIX   := $(shell ocamlc -config | awk -F '[\t \r]+' '/^ext_lib:/ {print $$2}' | tr -d '.')
 
 # If we are under Windows (regardless of whether we are using MSVC or mingw)
 # then the name of the executable file ends in .exe.
@@ -91,8 +86,9 @@
 # performed if "os_type" is "Win32" or "Win64", and must not be performed if
 # "os_type" is "Cygwin" or "Unix".
 
-ifneq (,$(shell ocamlc -config | grep -E "os_type: (Win32|Win64)"))
-installation_libdir := $(shell cygpath -m $(libdir))
+OS_TYPE:=	$(shell ocamlc -config | awk -F '[\t \r]+' '/^os_type:/ {print $$2}')
+ifeq ($(OS_TYPE),Win32)
+installation_libdir := $(shell cygpath -m $(libdir) || echo $(libdir))
 else
 installation_libdir := $(libdir)
 endif
--- ./src/cmly_write.ml
+++ ./src/cmly_write.ml
@@ -168,6 +168,6 @@
   output_value oc (t : grammar)
 
 let write filename =
-  let oc = open_out filename in
+  let oc = open_out_bin filename in
   write oc (encode());
   close_out oc
